node-src/lib/getE2eBinPath.ts

```diff
@@ -0,0 +1,18 @@
+import { Context } from '../types';
+import missingDependency from '../ui/messages/errors/missingDependency';
+import { exitCodes, setExitCode } from './setExitCode';
+import { failed } from '../ui/tasks/build';
+
+export function getE2eBinPath(ctx: Context, flag: 'playwright' | 'cypress') {
+  const dependencyName = `chromatic-${flag}`;
+  try {
+    return require.resolve(`${dependencyName}/bin/build-archive-storybook`);
+  } catch (err) {
+    if (err.code === 'MODULE_NOT_FOUND') {
+      ctx.log.error(missingDependency({ dependencyName, flag }));
+      setExitCode(ctx, exitCodes.MISSING_DEPENDENCY, true);
+      throw new Error(failed(ctx).output);
+    }
+    throw err;
+  }
+}
```

Comment:
Are we adding the new packages as optional peer dependencies in this PR?

Reply:
This is a good qn!

Reply:
The only piece that needs to be added here, based on scoped work, is the optional peer dependencies.  However, is it crucial for all of this to work?

Reply:
Only in some very specific cases but we may as well just do it! I will get to it